plugins/vars/sops.py

```diff
@@ -93,7 +92,9 @@ def get_vars(self, loader, path, entities, cache=True):
                         if os.path.exists(b_opath):
                             if os.path.isdir(b_opath):
                                 self._display.debug("\tprocessing dir %s" % opath)
-                                found_files = loader.find_vars_files(opath, entity.name, extensions=DEFAULT_VALID_EXTENSIONS)
+                                found_files = loader.find_vars_files(opath, entity.name)
+                                found_files = [file_path for file_path in found_files
+                                               if any(file_path.endswith(extension) for extension in DEFAULT_VALID_EXTENSIONS)]
```

Comment:
`any(to_text(file_path).endswith(extension) for extension in ...` should do the trick (note the added `to_text()`).